concrete/controllers/single_page/register.php

```diff
@@ -138,7 +146,12 @@ public function do_register()
         if (!$e->has()) {
             // do the registration
             $data = $_POST;
-            $data['uName'] = $username;
+            if ($this->displayUserName) {
+                $data['uName'] = $username;
+            } else {
+                $userService = new \Concrete\Core\Application\Service\User();
```

Comment:
Let's use $app->make(\Concrete\Core\Application\Service\User::class) instead so that this can be easily overridden